specification/2.0/Specification.adoc

```diff
@@ -1367,31 +1593,94 @@ The following example shows an image pointing to an external PNG image file and
 }
 ----
 
-[NOTE]
-.Implementation Note
-====
-When image data is provided by `uri` and `mimeType` is defined, client implementations should prefer JSON-defined MIME Type over one provided by transport layer.
-====
+Client implementations **MAY** need to manually determine the media type of some images. In such a case, the following table **SHOULD** be used to check the values of the first few bytes.
 
-The origin of the UV coordinates (0, 0) corresponds to the upper left corner of a texture image.
-This is illustrated in the following figure, where the respective UV coordinates are shown for all four corners of a normalized UV space:
+[options="header"]
+|====
+| Media Type   | Pattern Length | Pattern Bytes
+| `image/png`  | 8              | `0x89 0x50 0x4E 0x47 0x0D 0x0A 0x1A 0x0A`
+| `image/jpeg` | 3              | `0xFF 0xD8 0xFF`
+|====
+
+The image data **MUST** match the `image.mimeType` property when the latter is defined.
+
+The origin of the UV coordinates (0, 0) corresponds to the upper left corner of a texture image. This is illustrated in the following figure, where the respective UV coordinates are shown for all four corners of a normalized UV space:
 
 image:figures/texcoords.jpg[pdfwidth=4in,align=left]
 
-Any colorspace information (such as ICC profiles, intents, etc) from PNG or JPEG containers must be ignored. Effective transfer function is defined by a glTF object that refers to the image.
+Any colorspace information (such as ICC profiles, intents, gamma values, etc) from PNG or JPEG images **MUST** be ignored. Effective transfer function (encoding) is defined by a glTF object that refers to the image (in most cases it's a texture that is used by a material).
 
 [NOTE]
-.Implementation Note
+.Web Implementation Note
 ====
-This increases portability of an asset, since not all image decoding libraries fully support custom color conversions. To achieve correct rendering, WebGL runtimes must disable such conversions by setting `UNPACK_COLORSPACE_CONVERSION_WEBGL` flag to `NONE`.
+To ignore embedded colorspace information when using WebGL API, set `UNPACK_COLORSPACE_CONVERSION_WEBGL` flag to `NONE`.
+
+To ignore embedded colorspace information when using ImageBitmap API, set `colorSpaceConversion` option to `none`.
 ====
 
 
 [[samplers]]
 === Samplers
 
-Samplers are stored in the `samplers` array of the asset. Each sampler specifies filter and wrapping options corresponding to the GL types. The following example defines a sampler with linear mag filtering, linear mipmap min filtering, and repeat wrapping in S (U) and T (V).
+==== Overview
+Samplers are stored in the `samplers` array of the asset. Each sampler specifies filtering and wrapping modes.
+
+Client implementations **SHOULD** follow specified filtering modes. When the latter are undefined, client implementations **MAY** set their own default texture filtering settings.
+
+Client implementations **MUST** follow specified wrapping modes.
+
+==== Filtering
```

Comment:
There are some strong and difficult requirements mentioned here. Although it (only) says "**SHOULD** follow specified filtering mode", it ends with "client implementations **MUST** be able to generate mipmaps at runtime". 

It's quite a burden to put on clients in view of the broader context of glTF. The point is: Clients will just pass the magnification info to OpenGL/WebGL, and have no influence whatsoever on whether these constraints are met. They will be, usually, but that's essentially in the responsibility of the graphics card / driver vendors. 

tl;dr: Should (or can) the glTF spec dictate MipMap-lookup-rules?

(Related: A bit later, in the context of "non power-of-two textures", this question is somewhat offloaded by talking about the PLATFORMS that the clients are running on...)

Reply:
Please review the updated language in that section below.